### PR TITLE
Refactor and rewrite of the academy conversion project 

### PR DESCRIPTION
**Pull request by MartinWheelerMT**
Remove now unnecessary references to other data sources which are no longer required.
Add order to EF queries to retrieve most recent projects first
Update tests
General Refactoring.

### CODE REVIEW

File: TramsDataApi/UseCases/GetAcademyConversionProject.cs [hunk 1]
```diff
@@ -1,35 +1,32 @@
...
-            if (academyConversionProject == null) return null;
-
-            var trust = string.IsNullOrEmpty(academyConversionProject.TrustReferenceNumber)
-                ? null
-                : _trustGateway.GetIfdTrustByGroupId(academyConversionProject.TrustReferenceNumber);
-
-            return AcademyConversionProjectResponseFactory.Create(academyConversionProject, trust, null);
+            var academyConversionProject = _academyConversionProjectGateway.GetById(id);
+            if (academyConversionProject?.SchoolName == null) return null;
+
```
Comment (CathLass) [inline TramsDataApi/UseCases/GetAcademyConversionProject.cs:24]: Do we need to think about what these schoolName == null filters are about and whether we want them?

File: TramsDataApi/UseCases/GetAcademyConversionProjects.cs [hunk 2]
```diff
@@ -2,65 +2,36 @@
...
-            var responses = academyConversionProjects
-                .Where(p => !string.IsNullOrEmpty(p.SchoolName))
-                .Select(p => AcademyConversionProjectResponseFactory
-                    .Create(p, null, ifdProjects.FirstOrDefault(ifd => ifd.Sk == p.IfdPipelineId)))
-                .ToList();
-
+            var conversionProjects = _academyConversionProjectGateway.GetProjects(page, count);
+                      
+            var responses = conversionProjects
+                .Where(acp => acp.SchoolName != null)
```
Comment (CathLass) [inline TramsDataApi/UseCases/GetAcademyConversionProjects.cs:27]: Do we need to think about what these schoolName == null filters are about and whether we want them?

File: TramsDataApi/UseCases/GetAcademyConversionProjectsByStatuses.cs [hunk 3]
```diff
@@ -7,52 +7,32 @@
...
-                .Select(t => new {t.TrustRef, TrustName = t.TrustsTrustName})
-                .ToArray();
+                .GetByStatuses(page, count, statuses)
+                .ToList() ;
 
             var responses = academyConversionProjects
                 .Where(p => !string.IsNullOrEmpty(p.SchoolName))
-                .Select(p => AcademyConversionProjectResponseFactory
-                    .Create(p, null, ifdProjects.FirstOrDefault(ifd => ifd.Sk == p.IfdPipelineId)))
+                .Select(AcademyConversionProjectResponseFactory.Create)
```
Comment (CathLass) [inline TramsDataApi/UseCases/GetAcademyConversionProjectsByStatuses.cs:31]: another schoolName not null check

**Review by CathLass — APPROVED** [review]

File: TramsDataApi/UseCases/GetAcademyConversionProject.cs (hunk 1 above)
Comment (MartinWheelerMT) [reply TramsDataApi/UseCases/GetAcademyConversionProject.cs:24]: Will remove this, it appears that this was a workaround to ignore incomplete projects in an earlier iteration.  This should not happen with the new project creation approach

File: TramsDataApi/UseCases/GetAcademyConversionProjects.cs (hunk 2 above)
Comment (MartinWheelerMT) [reply TramsDataApi/UseCases/GetAcademyConversionProjects.cs:27]: Will remove this, it appears that this was a workaround to ignore incomplete projects in an earlier iteration.  This should not happen with the new project creation approach

**Pull request merged**